tools/test-proxy/Azure.Sdk.Tools.TestProxy/Common/ApplyCondition.cs

```diff
@@ -0,0 +1,142 @@
+﻿using System;
+using System.Linq;
+using System.Net;
+using System.Text;
+using System.Text.Json;
+using System.Text.RegularExpressions;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Http.Features;
+
+namespace Azure.Sdk.Tools.TestProxy.Common
+{
+    public class ApplyCondition
+    {
+
+        private string _uriRegex;
+        public string UriRegex { 
+            get
+            {
+                return _uriRegex;
+            }
+            set
+            {
+                Regex = new Regex(value);
+                _uriRegex = value;
+            }
+        }
+        private Regex Regex;
+
+        public ApplyCondition(string uriRegex)
+        {
+            UriRegex = uriRegex;
+        }
+
+        private JsonProperty GetProp(string name, JsonElement jsonElement)
+        {
+            return jsonElement.EnumerateObject()
+                        .FirstOrDefault(p => string.Compare(p.Name, "UriRegex", StringComparison.OrdinalIgnoreCase) == 0);
+        }
+
+        /// <summary>
+        /// This constructor is used to abstract the creation of an ApplyCondition from API input.
+        /// This is a separate function to allow context-sensitive setting. EG, if setting a condition,
+        /// one of the trigger properties should be populated! This is a bit immaterial when only dealing with 
+        /// a single property, but we might as well start this way.
+        /// </summary>
+        /// <param name="jsonElement">The contents of "condition" key with the body of a request to /Admin/AddSanitizer or /Admin/AddTransform.</param>
+        public ApplyCondition(JsonElement jsonElement)
+        {
+            var conditionDefined = false;
+            try
+            {
+                var uriProp = GetProp("UriRegex", jsonElement);
+
+                if(uriProp.Value.ValueKind != JsonValueKind.Undefined)
```

Comment:
Doesn't this need to be a string?

Reply:
We could key off of string only yes. Property.GetString() will work for bool items as well, so I was ok with just checking that it wasn't undefined, then forcing it. If we hit an error it'll be pretty explicit.